ui/apps/platform/src/Containers/Vulnerabilities/VulnerablityReporting/hooks/useDeleteModal.ts

```diff
@@ -0,0 +1,57 @@
+import useModal from 'hooks/useModal';
+import { useState } from 'react';
+import { deleteReportConfiguration } from 'services/ReportsService';
+import { getErrorMessage } from '../errorUtils';
+
+export type UseDeleteModalResult = {
+    openDeleteModal: (reportId: string) => void;
+    isDeleteModalOpen: boolean;
+    closeDeleteModal: () => void;
+    isDeleting: boolean;
+    onDelete: () => void;
+    deleteError: string | null;
+    isDeleted: boolean;
+};
+
+function useDeleteModal(): UseDeleteModalResult {
+    const { isModalOpen: isDeleteModalOpen, openModal, closeModal } = useModal();
+    const [reportIdToDelete, setReportIdToDelete] = useState<string>('');
+    const [isDeleting, setIsDeleting] = useState(false);
+    const [deleteError, setDeleteError] = useState<string | null>(null);
+    const [isDeleted, setIsDeleted] = useState(false);
```

Comment:
Considering the number of states, I could simplify the code by using some sort of state management model. Not sure if entirely necessary/overkill

Reply:
I think this case could go either way, but I wouldn't say it is necessary. If many of the states were changing at once in these functions I could see a `useReducer` clearing things up but that isn't really the case here.